tests/test_detector.py

```diff
@@ -55,7 +54,9 @@ def test_run(self):
             self.assertRaises(Exception, self.detector.run)
 
         with self.subTest("Test successful run"):
-            self.detector.run(DummyCheck(), DummyCheck())
+            check = Mock(spec=Check)
+            check.run.return_value = 0
+            self.detector.run(check, check)
             self.assertEqual(len(self.detector.check_reports), 2)
```

Comment:
you might also test, if the check was called

```suggestion
             self.assertEqual(len(self.detector.check_reports), 2)
             self.assertEqual(check.call_count, 2)
```